packages/admin/admin/src/error/errordialog/createErrorDialogApolloLink.tsx

```diff
@@ -20,48 +17,39 @@ export const createErrorDialogApolloLink = () => {
 
         let error: string | string[] | undefined;
         let errorType: ErrorType | undefined;
-        let title: React.ReactNode | undefined;
-        let userMessage: React.ReactNode | undefined;
         let httpStatus: string | undefined;
 
         if (graphQLErrors) {
             error = graphQLErrors.map(({ message }) => message);
-            errorType = "graphql";
-            title = <FormattedMessage id="comet.errorDialog.graphQLErrors.title" defaultMessage="Server error" />;
+            if (graphQLErrors.some((e) => e.message === "UNAUTHENTICATED")) {
+                errorType = "unauthenticated"; // Error is triggered by Comet Guard
+            } else if (
+                graphQLErrors.some((e) => e.extensions?.code === "UNAUTHENTICATED" || e.extensions?.exception?.status === StatusCodes.UNAUTHORIZED)
+            ) {
+                errorType = "unauthorized"; // Error is triggered by UnauthorizedException (which triggers an unauthenticated error code)
+            } else {
+                errorType = "graphql";
+            }
         }
 
         if (networkError) {
             // Prioritize GraphQL error over network error
             if (!error) {
                 error = networkError.message;
                 errorType = "network";
-                title = <FormattedMessage id="comet.errorDialog.networkError.title" defaultMessage="Network error" />;
             }
 
             if (isServerError(networkError)) {
                 const { statusCode } = networkError;
                 httpStatus = `${statusCode} ${getReasonPhrase(statusCode)}`;
-
                 if (statusCode === StatusCodes.UNAUTHORIZED) {
-                    title = <FormattedMessage id="comet.errorDialog.sessionExpired.title" defaultMessage="Session expired" />;
-                    userMessage = (
-                        <>
-                            <Typography gutterBottom>
-                                <FormattedMessage id="comet.errorDialog.sessionExpired.message" defaultMessage="Your login-session has expired." />
-                            </Typography>
-                            <Button href="/" color="info" variant="outlined">
-                                <FormattedMessage id="comet.errorDialog.sessionExpired.button" defaultMessage="Re-login" />
-                            </Button>
-                        </>
-                    );
+                    errorType = "unauthenticated"; // Error is triggered by AuthProxy
```

Comment:
The `UnauthorizedException` in line 29 triggers an unauthenticated error that should be shown as an unauthorized error.

And in line 45 the UNAUTHORIZED status code should be shown as an unauthenticated error.

Doesn't seem very intuitive. Can we improve that?

Reply:
very intuitive 🤦‍♂️